lib/plugins/rest.js

```diff
@@ -26,16 +37,27 @@ exports.register = function (server, options, next) {
 			var type = request.payload.type;
 			instances.createInstance(ami, type)
 			.then(function (instance) {
+				return awsAdapter.runInstances(instance);
+			})
+			.then(function (instance) {
 				reply(instance).created("/v1/instances/" + instance.id);
 			})
 			.catch(function (error) {
 				if (error.name === "ValidationError") {
 					reply("Bad Request: Please check the parameters passed.").code(400);
 				}
+				else if (error.name === "InvalidAMIID.Malformed") {
+					reply("The AMI entered does not exist. Ensure it is of the form ami-xxxxxx.").code(400);
+				}
+				else if (error.name === "InvalidParameterValue") {
+					reply("The Type entered does not exist. Ensure it is a valid EC2 type.").code(400);
+				}
+				else if (error.name === "AuthFailure") {
+					reply("Authentication Failure. Ensure your AWS credentials have been correctly used.").code(400);
```

Comment:
This isn't actually a 400, since the credentials are configured in the server. The error here is due to the server admin not starting the server with the correct configuration, so it should be a 500 error.